sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -160,7 +208,61 @@ private[spark] object CosmosClientCache extends BasicLoggingTrait {
   private[this] case class CosmosClientCacheMetadata
   (
     client: CosmosAsyncClient,
-    lastRetrieved: AtomicLong,
-    created: AtomicLong
-  )
+    lastModified: AtomicLong,
+    created: AtomicLong,
+    refCount: AtomicLong,
+    owners: TrieMap[OwnerInfo, Option[Boolean]]
+  ) {
+    def createCacheItemForReuse(ownerInfo: OwnerInfo) : CacheItemImpl = {
+      lastModified.set(Instant.now.toEpochMilli)
+      refCount.incrementAndGet()
+      owners.putIfAbsent(ownerInfo, None)
```

Comment:
The ownerInfo will always be unique for each task right? since it also include taskId, stageId etc